### PR TITLE
Feature: Article Renderer

### LINKED ISSUE #22: Feature: Article renderer
Create an article renderer to render the article on the website using data from the database

### PR DESCRIPTION
**Pull request by BrilliantDeviation7**
The article renderer displays a Markdown article with styling from Tailwind's typography plugin and also allows for HTML to embed videos either from self-hosted videos or from YouTube. The article is currently just a placeholder string on the page; in the future, it will be loaded from the database.

Resolves #22

image

### CODE REVIEW

File: src/app/_components/ArticleRenderer.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import rehypeRaw from "rehype-raw";
+
+interface MarkdownRendererProps {
+  content: string;
+}
+
+const MarkdownArticle: React.FC<MarkdownRendererProps> = ({ content }) => {
+  return (
+    <div className="prose">
+      <ReactMarkdown rehypePlugins={[rehypeRaw]}>{content}</ReactMarkdown>
```
Comment (elijah-wright) [inline src/app/_components/ArticleRenderer.tsx:11]: should we support [remark-gfm](https://github.com/remarkjs/remark-gfm)?

**Comment by alexng353** [comment]
Can we support [this](https://github.com/orgs/remarkjs/discussions/1043) way of embedding videos in markdown? It says we can modify how Remark processes the markdown by changing the syntax trees.

File: src/app/_components/ArticleRenderer.tsx (hunk 1 above)
Comment (BrilliantDeviation7) [reply src/app/_components/ArticleRenderer.tsx]: I added GitHub Flavored Markdown support.

**Comment by BrilliantDeviation7** [comment]
image

**Comment by alexng353** [comment]
Please add LaTeX support and resolve conflicts with /main.

**Pull request closed without merging**